flytekit-java/src/main/java/org/flyte/flytekit/SdkWorkflow.java

```diff
@@ -34,7 +36,22 @@ protected SdkWorkflow(SdkType<InputT> inputType, SdkType<OutputT> outputType) {
     this.outputType = outputType;
   }
 
-  public abstract void expand(SdkWorkflowBuilder builder);
+  public abstract OutputT expand(SdkWorkflowBuilder builder, InputT input);
+
+  protected void expand(SdkWorkflowBuilder builder) {
+    getInputType()
+        .getVariableMap()
+        .forEach(
+            (name, variable) ->
+                builder.inputOf(
+                    name,
+                    variable.literalType(),
+                    variable.description() == null ? "" : variable.description()));
+
+    OutputT output = expand(builder, getInputType().promiseFor(START_NODE_ID));
+
+    getOutputType().toSdkBindingMap(output).forEach(builder::output);
+  }
```

Comment:
```suggestion
  public final void expand(SdkWorkflowBuilder builder) {
    inputType
        .getVariableMap()
        .forEach(
            (name, variable) ->
                builder.inputOf(
                    name,
                    variable.literalType(),
                    variable.description() == null ? "" : variable.description()));

    OutputT output = expand(builder, inputType.promiseFor(START_NODE_ID));

    outputType.toSdkBindingMap(output).forEach(builder::output);
  }
```
We don't want subclasses to change the implementation an also before it was public, if `public` is not neccesary then it should changed to package private